### ISSUE TITLE
pytest suite does not run with Python 3.8

### ISSUE THREAD

**Issue by benclifford** [Parsl/parsl#1364]
**Describe the bug**

In PR #1363 which adds python 3.8 to CI:

```
$ pytest parsl -k "not cleannet" --config parsl/tests/configs/local_threads.py --cov=parsl --cov-append --cov-report= --random-order
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.8.0/bin/pytest", line 8, in <module>
    sys.exit(main())
  File "/home/travis/virtualenv/python3.8.0/lib/python3.8/site-packages/_pytest/config/__init__.py", line 58, in main
    config = _prepareconfig(args, plugins)
  File "/home/travis/virtualenv/python3.8.0/lib/python3.8/site-packages/_pytest/config/__init__.py", line 182, in _prepareconfig
    config = get_config()
  File "/home/travis/virtualenv/python3.8.0/lib/python3.8/site-packages/_pytest/config/__init__.py", line 153, in get_config
    pluginmanager.import_plugin(spec)
  File "/home/travis/virtualenv/python3.8.0/lib/python3.8/site-packages/_pytest/config/__init__.py", line 522, in import_plugin
    __import__(importspec)
  File "/home/travis/virtualenv/python3.8.0/lib/python3.8/site-packages/_pytest/tmpdir.py", line 25, in <module>
    class TempPathFactory(object):
  File "/home/travis/virtualenv/python3.8.0/lib/python3.8/site-packages/_pytest/tmpdir.py", line 30, in TempPathFactory
    _given_basetemp = attr.ib(
TypeError: attrib() got an unexpected keyword argument 'convert'
The command "pytest parsl -k "not cleannet" --config parsl/tests/configs/local_threads.py --cov=parsl --cov-append --cov-report= --random-order" exited with 1.
```

Our test-requirements restricts pytest to quite an old version: <4.1.0, when at time of writing recent pytests are 5.2.1 and 4.6.6

**To Reproduce**
Install python 3.8. Run the above pytest command.

**Expected behavior**
Tests should pass.

**Environment**
CI

**Comment by benclifford** [comment]
also see #1415 which attempts to update pytest to avoid this problem

**Comment by benclifford** [comment]
Fixed by updating pytest in #1415
